### ISSUE TITLE
Water bottles don't restore hydration

### ISSUE THREAD

**Issue by axlegear** [EnviroMine/EnviroMine-1.7#211]
Cooking dirty water bottles to clean, they do nothing.

**Comment by PitchBright** [comment]
Please try to describe the issue more clearly. You're not offering up very much information or detail.

**Comment by axlegear** [comment]
If you fill empty bottles with dirty water, then cook them in a furnace, it makes water bottles.  These water bottles do not restore any hydration/thirst or affect temperature or anything.  They do nothing.
Can't think of any way to more clearly describe that they don't have any effect.

**Comment by PitchBright** [comment]
Okay... your title was "Water bottles don't restore hydration" (which I understood to mean Water Bottles aren't hydrating the player.)

Then you describe the problem completely differently by saying that something that sounded like you couldn't cook dirty water bottles into clean.

**Comment by PitchBright** [comment]
Are you saying that Clean Water bottles collected from a water source work (hydrate the player)... but that Clean Water bottles cooked from Dirty Water bottles don't work?

**Comment by axlegear** [comment]
The last one..  "but that Clean Water bottles cooked from Dirty Water bottles don't work?"

**Comment by PitchBright** [comment]
(It wasn't an either/or question.)

So clean water collected from a lake works then?

**Comment by axlegear** [comment]
Yes, only cooked water is broken.

Sorry if I am hard to understand.

**Comment by PitchBright** [comment]
See if you can figure out what the difference is between Clean Water Bottle (collected from lake) and Clean Water Bottle (made from cooking Dirty Water in Furance) is. That will go a long way in helping you figure out if you found a bug with the mod.

**Comment by axlegear** [comment]
I would not know how to do this.  I am not a modder or dev or programmer.

**Comment by PitchBright** [comment]
If you are using NEI, or some other type of "more info" tool/mod... you can hover over items and see more information about them (name, id#, etc).

Gather some clean water from a lake. See the info on it.
Make some clean water from furnace. See the info on it.

Report back here with your findings :)

**Comment by axlegear** [comment]
![image](https://cloud.githubusercontent.com/assets/7531754/17671676/e41710b2-62dd-11e6-9298-ecc0a10cd0b0.png)
The above is dirty water cooked

![image](https://cloud.githubusercontent.com/assets/7531754/17671729/1f3aa726-62de-11e6-8029-67c159f56e11.png)
This is from fresh water

**Comment by axlegear** [comment]
The fresh water worked fine but the dirty water I cooked doesn't

**Comment by PitchBright** [comment]
Bizarre. They look identical. I can't imagine why one would work, and the other wouldn't.

**Comment by axlegear** [comment]
NBT data?  No idea, there must be some unviewable attached info

**Comment by PitchBright** [comment]
I'm thinkin' the same thing as you.

**Comment by PitchBright** [comment]
When you have the clean lake water in your hotbar... and you shift click on the clean water in the furnace output slot... does it stack into the clean lake water in your hotbar?

Edit: Will they stack together at all, I guess.

**Comment by axlegear** [comment]
no

**Comment by PitchBright** [comment]
Yeah, something might be wrong then. Leave this issue open for now.

**Comment by axlegear** [comment]
Oho!  Interesting fix.  Someone noted it works on singleplayer but not the server.
I checked and the server doesn't generate any configs under minecraft\config\enviromine\profiles\default\CustomProperties
but SP does

So I copied the SP configs to the server, and now it all works
It seems this file has something relevent:

[enviromine.txt](https://github.com/EnviroMine/EnviroMine-1.7/files/418552/enviromine.txt)

**Comment by axlegear** [comment]
So if anyone else has this issue, the above is a workaround fix
Tho still a bug with the mod not generating proper configs for server so that needs fixing

**Comment by PitchBright** [comment]
Awesome detective work, @axlegear !

That also explains why I never had any of these issues. I generated my configs in SP, and then moved them all to the server, when I was ready to run the server with EM.

**Comment by axlegear** [comment]
Hopefully at least it will help the modders find and fix the issue.  Prolly pretty simple
